spec/integration/sdam_error_handling_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'spec_helper'
+
+describe 'SDAM error handling' do
+  before(:all) do
+    ClientRegistry.instance.close_all_clients
+  end
+
+  describe 'when there is an error during an operation' do
+    let(:client) { authorized_client }
+
+    before do
+      wait_for_all_servers(client.cluster)
+      # we also need a connection to the primary so that our error
```

Comment:
It's not clear to me at all what this comment has to do with the code after it; what does needing a connection to the primary have to do with pinging the server and stopping the monitors?

Reply:
Ping establishes a connection to the primary, the comment goes with the ping. Stopping the monitors is separate. Ultimately this is all done to ensure assertions are performed on the right connections (non-monitoring vs monitoring).